### PR TITLE
Parent pom enhancements plus jsf

### PR DESCRIPTION
**Pull request by ngriffin7a**
Since the maven examples in the blade project are demos/tests (not archetypes), the duplicated child pom information has been consolidated into the parent pom. Additionally, the beginnings of a JSF demo is included.

### CODE REVIEW

**Comment by ngriffin7a** [comment]
@gamerson I just updated my branch with commit https://github.com/ngriffin7a/blade/commit/cb8f4917bd6e1e7793293838b9331ca7503770f8

**Comment by gamerson** [comment]
Hey @ngriffin7a 

Thanks a ton for the pull.  One quick thing.  We had decided previously to not introduce a parent pom for maven projects.  I know that is an anti-pattern, but we decided that we wanted all of the blade samples to be completely self-contained.  So a developer can copy&paste a single folder and it can be copied to whereever and built automatically. 

Our thinking may change in the future, but for now can you resend this without the maven parent stuff?  

Thanks, closing for now.

File: .gitignore [hunk 1]
```diff
@@ -4,3 +4,6 @@ bin
 reports/
 generated/
 .gradle
+*.iml
```
Comment (gamerson) [inline .gitignore]: @ngriffin7a who is using IntelliJ? haha.

**Comment by gamerson** [comment]
Merged here: https://github.com/rotty3000/blade/commit/3521f0b4f7b438a1a53c5950bcbb3b8fa2a96bdb

**Pull request closed without merging**